src/Runner.Common/RunServer.cs

```diff
@@ -67,10 +67,48 @@ private void CheckConnection()
             }
         }
 
-        public Task<AgentJobRequestMessage> GetJobMessageAsync(string id)
+        public Task<AgentJobRequestMessage> GetJobMessageAsync(string id, CancellationToken cancellationToken)
         {
             CheckConnection();
-            return _taskAgentClient.GetJobMessageAsync(id);
+            var jobMessage = RetryRequest<AgentJobRequestMessage>(async () =>
+                                                    {
+                                                        return await _taskAgentClient.GetJobMessageAsync(id, cancellationToken);
+                                                    }, cancellationToken);
+            return jobMessage;
+        }
+
+        private async Task<T> RetryRequest<T>(Func<Task<T>> func,
+                                                        CancellationToken cancellationToken,
+                                                        int maxRetryAttempts = 5
+                                                        )
+        {
+            var currentRetryAttempt = 1;
+            while (true)
+            {
+                cancellationToken.ThrowIfCancellationRequested();
+                try
+                {
+                    return await func();
+                }
+                catch (Exception ex)
```

Comment:
We could catch if `retryCount / currentRetryAttempt` is not too high, otherwise let it bubble through.

`catch(Exception ex) when (retryCount < maxRetryAttempt)`